classes/Page.php

```diff
@@ -244,18 +244,19 @@ public function delete()
         foreach ($this->getChildren() as $subPage) {
             $result = array_merge($result, $subPage->delete());
         }
```

Comment:
@LukeTowers should this block perhaps execute after `parent::delete()` as well, if we're allowing people to halt the process?

Reply:
I thought about that but was concerned about edge cases breaking things if the parent was fully gone before the children were and didn't really have time to test that. Feel free to look into it further if it interests you though. I might revisit later, this was to support a "page specific editing permission" feature for specific guest users to the backend, so for a proper implementation you'd want to delete child pages only after confirming that the parent page can be deleted. 

It does get complicated though, what happens if a child page deletion fails? If the parent is already deleted then any siblings after the failing child would now be orphaned. Because of the potential issues I just decided that if there's anything that you would want to halt both you'd be responsible for doing that checking yourself rather than trying to work around these potential issues in a one-size-fits-all approach.

Reply:
I think it's safer to have orphaned child pages to deleted pages - at least they are recoverable. :)

Reply:
Ah yes, you are correct after thinking about it a bit more. Feel free to submit a PR with the change so you can count it towards Hacktoberfest :)

Reply:
Gunna gets me a shirt! :P